common/formatting/align.h

```diff
@@ -305,6 +308,68 @@ std::vector<ColumnPositionEntry> ScanPartitionForAlignmentCells(
   return scanner.SparseColumns();
 }
 
+// Also uses a non-tree column scanning function, whose columns are
```

Comment:
Probably a bit more detailed documentation needed so that if we look at it in half a year we still know what this is all about :)

Describe what the parameters are doing similar to above (_a row correspondents ..._) but also of course what the `non_tree_column_scanner` callback is , what it is expected to do etc.

Reply:
Added comment